src/libAtomVM/context.c

```diff
@@ -89,3 +90,46 @@ void context_destroy(Context *ctx)
     free(ctx->heap_start);
     free(ctx);
 }
+
+struct TermHandle *context_create_term_handle(Context *ctx, term t)
+{
+    struct TermHandle *handle = (struct TermHandle *) malloc(sizeof(struct TermHandle));
+    if (UNLIKELY(handle == NULL)) {
+        fprintf(stderr, "Memory allocation failure: %s:%i\n", __FILE__, __LINE__);
+        abort();
+    }
+    handle->t = t;
+    linkedlist_append(&ctx->term_handles, &handle->list_head);
+    return handle;
+}
+
+void context_release_term_handle(Context *ctx, struct TermHandle *handle)
+{
+    if (ctx->term_handles == NULL) {
+        return;
+    }
+    struct TermHandle *handles = GET_LIST_ENTRY(ctx->term_handles, struct TermHandle, list_head);
+    struct TermHandle *h = handles;
+    do {
+        if (h == handle) {
+            linkedlist_remove(&ctx->term_handles, &h->list_head);
```

Comment:
There is a bug here -- it should free the memory associated with the handle.  So technically this is a leak.  Will fix.